lib/libzfs/libzfs_sendrecv.c

```diff
@@ -893,9 +907,10 @@ send_iterate_fs(zfs_handle_t *zhp, void *arg)
 	}
 
 	/* iterate over props */
-	VERIFY(0 == nvlist_alloc(&nv, NV_UNIQUE_NAME, 0));
-	send_iterate_prop(zhp, sd->backup, nv);
-
+	if (sd->props || sd->backup || sd->recursive) {
```

Comment:
nit: while "recursive" unfortunately implies "replication" (-R) one could argue we don't need to send properties if we are only "recursive". That is to say, if a simpler "-r" (recursive only, no replication) option is going to be introduced in future versions we don't want to call `send_iterate_prop()` here.

Reply:
True.  But to to keep things working as they were before my changes I think we still need the || sd->recursive.

Reply:
I believe you are right, i tried running "zfs_send" and "zfs_receive" test groups with the following change applied and found some unexpected failures ("receive-o-x_props_override" and "zfs_receive_to_encrypted"):

```diff
diff --git a/lib/libzfs/libzfs_sendrecv.c b/lib/libzfs/libzfs_sendrecv.c
index e993b19..2d537b1 100644
--- a/lib/libzfs/libzfs_sendrecv.c
+++ b/lib/libzfs/libzfs_sendrecv.c
@@ -907,7 +907,7 @@ send_iterate_fs(zfs_handle_t *zhp, void *arg)
        }
 
        /* iterate over props */
-       if (sd->props || sd->backup || sd->recursive) {
+       if (sd->props || sd->backup) {
                VERIFY(0 == nvlist_alloc(&nv, NV_UNIQUE_NAME, 0));
                send_iterate_prop(zhp, sd->backup, nv);
        }
```

This could be a bug, i expected `sd->props` to be set when executing `zfs send -R` but maybe i was wrong; for the time being we should probably keep using `|| sd->recursive` like you suggested.